internal/clients/nightfall/nightfall.go

```diff
@@ -262,19 +283,52 @@ func (n *Client) scanAllContent(ctx context.Context, logger logger.Logger, cts [
 }
 
 // Scan send /scan request to Nightfall API and return findings
-func (n *Client) Scan(ctx context.Context, logger logger.Logger, items []string) ([][]nightfallAPI.ScanResponse, error) {
+func (n *Client) Scan(
+	ctx context.Context,
+	logger logger.Logger,
+	items []string,
+) ([][]nightfallAPI.ScanResponse, error) {
 	APIKey := nightfallAPI.APIKey{
 		Key:    n.APIKey,
 		Prefix: "",
 	}
 	newCtx := context.WithValue(ctx, nightfallAPI.ContextAPIKey, APIKey)
 	request := n.createScanRequest(items)
-	resp, _, err := n.APIClient.ScanAPI().ScanPayload(newCtx, request)
-	if err != nil {
-		logger.Error(fmt.Sprintf("Error from Nightfall API, unable to successfully scan %d items", len(items)))
-		return nil, err
+	return n.makeScanRequest(newCtx, logger, request)
+}
+
+func (n *Client) makeScanRequest(
```

Comment:
nit - can we update this name to fully reflect its functionality, e.g. `makeScanRequestWithRetries` or something?